Test Failure (20180420-0319): LoadTest.testCommitAndRollback RuntimePermission denied for WSJdbcTracer invoking newProxyInstance

Failure with Java 2 security enabled:
Test Failure (20180420-0319): com.ibm.ws.jdbc.loadtests.tests.LoadTest.testCommitAndRollback

junit.framework.AssertionFailedError: 2018-04-20-08:08:00:132 [Unexpected FFDC reporting java.security.AccessControlException was found (count = 119): /home/jazz_build/_NTiEaURXEei6Isn3iLaRtw-EBC.PROD.WASRTC-AHM-000-00-00/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.jdbc.loadtest/logs/ffdc/ffdc_18.04.20_08.07.52.0.log
>------Start of DE processing------ = [4/20/18 8:07:52:239 UTC]
>Exception = java.security.AccessControlException
>Source = com.ibm.ejs.j2c.poolmanager.FreePool.createManagedConnectionWithMCWrapper
>probeid = 545
>Stack Dump = java.security.AccessControlException: Access denied ("java.lang.RuntimePermission" "getClassLoader")
>	at java.security.AccessController.throwACE(AccessController.java:157)
>	at java.security.AccessController.checkPermissionHelper(AccessController.java:217)
>	at java.security.AccessController.checkPermission(AccessController.java:349)
>	at java.lang.SecurityManager.checkPermission(SecurityManager.java:562)
>	at java.lang.reflect.Proxy.checkProxyAccess(Proxy.java:411)
>	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:724)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcTracer.invoke(WSJdbcTracer.java:198)
>	at com.sun.proxy.$Proxy8.getConnection(Unknown Source)
>	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnection(WSManagedConnectionFactoryImpl.java:691)
>	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnection(WSManagedConnectionFactoryImpl.java:798)
>	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.createManagedConnection(WSManagedConnectionFactoryImpl.java:651)
>	at com.ibm.ejs.j2c.FreePool.createManagedConnectionWithMCWrapper(FreePool.java:1354)
>	at com.ibm.ejs.j2c.FreePool.createOrWaitForConnection(FreePool.java:1228)
>	at com.ibm.ejs.j2c.PoolManager.reserve(PoolManager.java:1434)
>	at com.ibm.ejs.j2c.ConnectionManager.allocateMCWrapper(ConnectionManager.java:573)
>	at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:309)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:143)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:116)
>	at web.LoadTestServlet.setUp(LoadTestServlet.java:92)
>	at web.LoadTestServlet.doGet(LoadTestServlet.java:69)
>	at javax.servlet.http.HttpServlet.service(HttpServlet.java:575)
>	at javax.servlet.http.HttpServlet.service(HttpServlet.java:668)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
>	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1154)
>	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:4962)
>	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
>	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:996)
>	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1011)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:414)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:373)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:530)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:464)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:329)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:300)
>	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:165)
>	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:74)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:501)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:571)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:926)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1015)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1153)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
>	at java.lang.Thread.run(Thread.java:785)
, Unexpected FFDC reporting javax.resource.spi.ResourceAllocationException was found (count = 119): /home/jazz_build/_NTiEaURXEei6Isn3iLaRtw-EBC.PROD.WASRTC-AHM-000-00-00/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.jdbc.loadtest/logs/ffdc/ffdc_18.04.20_08.07.52.1.log
>------Start of DE processing------ = [4/20/18 8:07:52:273 UTC]
>Exception = javax.resource.spi.ResourceAllocationException
>Source = com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection
>probeid = 299
>Stack Dump = javax.resource.spi.ResourceAllocationException: Access denied ("java.lang.RuntimePermission" "getClassLoader")
>	at com.ibm.ejs.j2c.FreePool.createManagedConnectionWithMCWrapper(FreePool.java:1576)
>	at com.ibm.ejs.j2c.FreePool.createOrWaitForConnection(FreePool.java:1228)
>	at com.ibm.ejs.j2c.PoolManager.reserve(PoolManager.java:1434)
>	at com.ibm.ejs.j2c.ConnectionManager.allocateMCWrapper(ConnectionManager.java:573)
>	at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:309)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:143)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:116)
>	at web.LoadTestServlet.setUp(LoadTestServlet.java:92)
>	at web.LoadTestServlet.doGet(LoadTestServlet.java:69)
>	at javax.servlet.http.HttpServlet.service(HttpServlet.java:575)
>	at javax.servlet.http.HttpServlet.service(HttpServlet.java:668)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
>	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
>	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1154)
>	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:4962)
>	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
>	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:996)
>	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1011)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:414)
>	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:373)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:530)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:464)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:329)
>	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:300)
>	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:165)
>	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:74)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:501)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:571)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:926)
>	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1015)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1153)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
>	at java.lang.Thread.run(Thread.java:785)
>Caused by: java.security.AccessControlException: Access denied ("java.lang.RuntimePermission" "getClassLoader")
>	at java.security.AccessController.throwACE(AccessController.java:157)
>	at java.security.AccessController.checkPermissionHelper(AccessController.java:217)
>	at java.security.AccessController.checkPermission(AccessController.java:349)
>	at java.lang.SecurityManager.checkPermission(SecurityManager.java:562)
>	at java.lang.reflect.Proxy.checkProxyAccess(Proxy.java:411)
>	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:724)
>	at com.ibm.ws.rsadapter.jdbc.WSJdbcTracer.invoke(WSJdbcTracer.java:198)
>	at com.sun.proxy.$Proxy8.getConnection(Unknown Source)
>	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnection(WSManagedConnectionFactoryImpl.java:691)
>	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnection(WSManagedConnectionFactoryImpl.java:798)
>	at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.createManagedConnection(WSManagedConnectionFactoryImpl.java:651)
>	at com.ibm.ejs.j2c.FreePool.createManagedConnectionWithMCWrapper(FreePool.java:1354)
]
	at componenttest.custom.junit.runner.FATRunner.blowup(FATRunner.java:277)
	at componenttest.custom.junit.runner.FATRunner.access$500(FATRunner.java:63)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:227)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:314)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:153)

--- a/dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/jdbc/WSJdbcTracer.java
+++ b/dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/jdbc/WSJdbcTracer.java
@@ -22,6 +22,8 @@ import java.lang.reflect.Method;
 import java.lang.reflect.Proxy;
 import java.math.BigDecimal;
 import java.net.URL;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
 import java.sql.CallableStatement;
 import java.sql.Connection;
 import java.sql.DatabaseMetaData;
@@ -195,7 +197,12 @@ public class WSJdbcTracer implements InvocationHandler
             // - include SQL for PreparedStatement & CallableStatement
             final String query = methName.startsWith("prepare") && PreparedStatement.class.isAssignableFrom(returnType)
                            && args.length >= 1 && args[0] instanceof String ? (String) args[0] : null;
-            traceableResult = Proxy.newProxyInstance(WSJdbcWrapper.priv.getClassLoader(returnType), new Class[] { returnType }, new WSJdbcTracer(tracer, writer, result, returnType, query, false));
+            final WSJdbcTracer newTracer = new WSJdbcTracer(tracer, writer, result, returnType, query, false);
+            traceableResult = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                public Object run() {
+                    return Proxy.newProxyInstance(returnType.getClassLoader(), new Class[] { returnType }, newTracer);
+                }
+            });
         }
 
         // Trace exit after successful invocation.

